packages/framework/src/Framework/Features/Navigation/NavigationManager.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Hyde\Framework\Features\Navigation;
+
+use Hyde\Hyde;
+
+/**
+ * Manages the navigation menus for the project.
+ */
+class NavigationManager
```

Comment:
This is basically a second service container with extra steps. Maybe it's easier to just bind the menu classes directly? (Unless we want to store the menus in the kernel, which we probably don't)

Reply:
It depends on the level of flexibility we want. Do we want each specific menu to be in its own class, or do we allow anonymous menus that are just instances of the NavigationMenu class?

Reply:
Going to experiment with the former

Reply:
See https://github.com/hydephp/develop/pull/1580

Reply:
A benefit with the class based approach is that it's less ambiguous than "identifiers"
![image](https://github.com/hydephp/develop/assets/95144705/d29555cd-4269-4476-8487-36c77f53c828)